### PR TITLE
AWS: Consolidates Lambda::Permission objects for cloudwatchLog events

### LINKED ISSUE #5357: The final policy size is bigger than the limit (20480)
# This is a Bug Report

## Description

* What went wrong? **Stack update failed. Removing and redeploying did not help.**
* What did you expect should have happened? **Stack update should complete successfully.**
* What was the config you used? **serverless.yml contains a single function with dozens of event sources as subscription filters to CloudWatch Logs.**
* What stacktrace or error message from your provider did you see? `The final policy size (size here differs between deployments) is bigger than the limit (20480)`

Similar or dependent issues:
* None?

## Additional Data

* ***Serverless Framework Version you're using***: 1.32.0
* ***Operating System***: Ubuntu 18.04
* ***Stack Trace***:
* ***Provider Error messages***: 
```
  Serverless Error ---------------------------------------
 
  An error occurred: SubscriptionFilterLambdaLambdaPermissionLogsSubscriptionFilterCloudWatchLog28 - The final policy size (20623) is bigger than the limit (20480). (Service: AWSLambda; Status Code: 400; Error Code: PolicyLengthExceededException; Request ID: a87d245a-c85c-11e8-a18a-af6f9fead152).
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information -----------------------------
     OS:                     linux
     Node Version:           8.9.1
     Serverless Version:     1.32.0
```
The issue seems to have been reported on the [forums](https://forum.serverless.com/t/aws-lambda-permission-policy-size/3952) however the linked solution does not work.

### PR DESCRIPTION
**Pull request by exoego**
## What did you implement:
Closes #5357

## How did you implement it:

<del>Instead of creating `Permission` object separately for every `cloudwatchLog` events,
one-and-only `Permission` is created and shared with `cloudwathLog`-subsciber functions in one service `yml`.</del>
Instead of creating `Permission` object separately for every `cloudwatchLog` events,
one-and-only `Permission` is created per function and shared with `cloudwathLog` events.

## How can we verify it:

1. Create service with many functions and cloudwatchLog subscriber

```yml
service: service-for-verification
provider:
  name: aws
  runtime: nodejs8.10
functions:
  hello1:
    handler: handler.hello1
  hello2:
    handler: handler.hello2
   ...
  hello40:
    handler: handler.hello40
  subscriber:
    handler: handler.subscriber
    events:
     - cloudwatchLog: "/aws/lambda/hello1"
     - cloudwatchLog: "/aws/lambda/hello2"
    ...
     - cloudwatchLog: "/aws/lambda/hello40"
```

2. Deploy it

## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** YES

# Note:

If we deploy on top of this PR, Lambda console will show warning message like below:
```
'cloudwatch-logs é¢ä¿ããªã¹ãããéã«ã¨ã©ã¼ãçºçãã¾ããï¼2 validation errors
 detected: Value '' at 'logGroupName' failed to satisfy constraint: Member 
must satisfy regular expression pattern: [\.\-_/#A-Za-z0-9]+; Value '' at
 'logGroupName' failed to satisfy constraint: Member must have length
 greater than or equal to 1 (Service: AWSLogs; Status Code: 400; Error Code:
 InvalidParameterException; Request ID: ....)
```

According to AWS support, this is cosmetic error and negligible.

### CODE REVIEW

**Comment by Jaystified** [comment]
Applied this change on top of 1.32.0 and my previously undeployable stack seems to deploy properly and work well. 
Hopefully this gets merged soon.

File: lib/plugins/aws/package/compile/events/cloudWatchLog/index.js [hunk 1]
```diff
@@ -120,9 +119,7 @@ class AwsCompileCloudWatchLogEvents {
                   { "Ref": "AWS::Region" },
                   ":",
                   { "Ref": "AWS::AccountId" },
-                  ":log-group:",
-                  "${LogGroupName}",
-                  ":*"
+                  ":log-group:*:*"
```
[the remark this answers is not in the document]
Comment (exoego) [reply lib/plugins/aws/package/compile/events/cloudWatchLog/index.js]: Done in https://github.com/serverless/serverless/pull/5531/commits/6d0cbe3ced97ed4d4a5ed1ec2304d8fe661e41aa and https://github.com/serverless/serverless/pull/5531/commits/7dadcdd422ac975643c3572617d614a7550cdc02

**Comment by exoego** [comment]
@dschep 
If I understand AWS correctly, I suppose your concern is more than necessary.

Yes, *implicit wildcard* may give unnecessarily-wide pattern like `/*` to `LogGroup`.
But `LogGroup` cannot `InvokeFunction` unless it has `SubscriptionFilter`.
And, according to AWS document, `LogGroup` has a hard-limit ["Subscription filters | 1 per log group. This limit cannot be changed"](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/cloudwatch_limits_cwl.html)

A `SubscriptionFilter` attached to `LogGroup` is the one which an user like me is going to deploy via `serverless.yml`.
And the `SubscriptionFilter` Lambda is allowed limited operations via `provider.iamRoleStatements`.
So... I think *implicit wildcard* does not become a security concern.

Please correct me if I miss something.
If un-configurability is unfit to design policy of Severless Framework, I am OK to update PR with *explicit wildcard* via `permissionPattern`.

**Pull request merged**